### PR TITLE
Initial MAAS ctl command

### PR DESCRIPTION
**Pull request by mikevanhemert**

### CODE REVIEW

**Review by docandrew — APPROVED** [review]
So happy for the foundation laid out here.

**Review by jamesallen-atx — APPROVED** [review]

File: bin/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+sleep 10
+
+echo "Executing Zarf init"
+zarf init --set REGISTRY_PVC_SIZE="128Gi" --components git-server --confirm
+
+count=1
+
+while read -r package; do
+  echo "${count} Deploying ${package}"
+  zarf p d ${working}/${package} --confirm
```
Comment (bburky) [inline bin/deploy.sh:52]:
```suggestion
  zarf package deploy "${working}/${package}" --confirm
```

File: bin/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+set -euo pipefail
+
+working=/tmp/pipelinetemp/
```
Comment (bburky) [inline bin/deploy.sh:5]: Could this be `mktemp -d` instead? It looks like we do an `rm -rf` on the dir at some point, so it really is just a temporary working directory? Ideally this is inside the pipeline's working directory so it gets cleaned up by the GH runner when done.

Or is it intended that this is accessible on the host outside the pipeline after it finishes? That won't work well if this switches to a k8s GH runner.

File: .github/workflows/end-to-end.yaml [hunk 3]
```diff
@@ -25,3 +27,70 @@ jobs:
...
+              mctl:0.1.0 -- \
+              machines ls
+      
+      - name: Release ATAR5
+        run: |
+          docker run \
+              -e MAAS_SERVER=${{secrets.MAAS_SERVER}} \
+              -e MAAS_API_KEY=${{secrets.MAAS_API_KEY}} \
+              mctl:0.1.0 -- \
+              machines release ATAR5
```
[the remark this answers is not in the document]
Comment (jamesallen-atx) [reply .github/workflows/end-to-end.yaml:51]: Added tags in MaaS and resource group. Updating the cli to include options for releasing resources based on owner, tag, or resource group

**Pull request merged**